src/schedule/mod.rs

```diff
@@ -225,15 +366,15 @@ impl WelcomeSecret {
 /// An intermediate secret in the key schedule, the `EpochSecret` is used to
 /// create an `EpochSecrets` object and is finally consumed when creating that
 /// epoch's `InitSecret`.
-pub(crate) struct EpochSecret {
+struct EpochSecret {
     secret: Secret,
 }
 
 impl EpochSecret {
     /// Derive an `EpochSecret` from a `JoinerSecret`
-    pub(crate) fn new(
+    fn new(
         ciphersuite: &Ciphersuite,
-        intermediate_secret: IntermediateSecret,
+        intermediate_secret: &IntermediateSecret,
```

Comment:
Can't we consume the intermediate secret here? I believe that should be possible if we use `take()` in `add_context`.